### ISSUE TITLE
toHaveStyle doesn't take account of CSS custom properties

### ISSUE THREAD

**Issue by thomaslombart** [testing-library/jest-dom#280]
- `@testing-library/jest-dom` version: 5.1.1
- `node` version: 12.16.2
- `npm` (or `yarn`) version: 1.22.4

### Relevant code or config:

```js
test.only('failing test', () => {
  const style = document.createElement('style')
  style.innerHTML = `
    div {
      --color: blue;
    }
  `

  const {container} = render(`
    <div>
      Hello world
    </div>
  `)

  document.body.appendChild(style)
  document.body.appendChild(container)

  expect(container).toHaveStyle(`--color: blue`)
})
```

### Problem description:

If you were to run the test above, this would fail because `toHaveStyle` doesn't take account of CSS custom properties.

### Suggested solution:

I think the PR #276 caused this behavior. Indeed, the `isSubset` function has been changed and doesn't use `getPropertyValue` anymore (see [this line](https://github.com/testing-library/jest-dom/pull/276/files#diff-4e1129d165c9efdd245d6f1726948bbbL24)).

It seems a custom property on [`CSSStyleDeclaration`](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleDeclaration) can't be accessed simply using the property key (for example `computedStyle['--color']`), you explicitely have to use [`getPropertyValue`](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleDeclaration/getPropertyValue).

I forked the repo and changed back the `isSubset` function to use `getPropertyValue` and things work well. That may be a solution (I assume there's a reason why it's been changed so things may not be so simple).

**Comment by gnapse** [comment]
Indeed, #276 was there for a reason, and unfortunately, we had not been covering in a test the support for these custom properties, which would've flagged this when it was introduced. The reason for #276, you can see it by checking #272, which is pretty serious and un-intuitive.

Anyway, we need to figure out how to support both use-cases. I may be able to dig into this soon, but I'll also mention here @just-boris who flagged #272 and authored the fix, because maybe they have an immediate understanding of what could be needed here.

**Comment by just-boris** [comment]
I see two possible options here:

1. Bring back the getPropertyValue call, while keeping the property getter:

```
Object.entries(styles).every(
      ([prop, value]) =>
        computedStyle[prop] === value ||
        computedStyle.getPropertyValue(prop.toLowerCase()) === value,
    )
```

2. Use only `getPropertyValue` and write a custom camelCase to dash-case converter, based on the following code: https://github.com/jsdom/cssstyle/blob/master/lib/parsers.js#L711-L722 (sadly, it is not exported)

**Comment by gnapse** [comment]
Option (1) seems reasonable enough without the extra overhead of the case converter.

**Comment by gnapse** [comment]
BTW @thomlom in the mean time, while this is resolved, you need not have a fork. You can fix the version in your package.json to the one prior to the release of that recent fix.

**Comment by thomlom** [comment]
Option 1 seems good to me too! Tests with a custom property would need to be added of course.

@gnapse, sure! I didn't mean "fork" but "cloned" 😄

**Comment by AshConnolly** [comment]
Can we get this issue reopened as it doesn't appear to have been resolved.
see here - https://github.com/testing-library/jest-dom/issues/322#issuecomment-988212751
PR kindly opened by @hughes-ch here - https://github.com/jsdom/jsdom/pull/3299

**Comment by gnapse** [comment]
If the fix lies on jsdom, then the fix on our side is to merely update the jsdom dependency, right? Which, BTW, may require an update to the jsdom dependency in dom testing library.

Anyway, I'll open this for now to reflect this situation.

**Comment by swese44** [comment]
Any update here? Thanks!

**Comment by gnapse** [comment]
Not yet. I wonder if updating the jsdom dependency could help. It is at v16.x here, and it could be taken to v20 like in `dom-testing-library`. I'll give it a try later, but no hard promises on the deadline. If someone wants to give it a try and ping me in the PR, I'll be happy to check it out.

**Comment by Fullchee** [comment]
Not ideal but the workaround I found (which might be helpful in figuring out the issue) was this

```js
function App() {
  return <div style={{ "--my-property": 1 }}>Hello</div>;
}

expect(app.style._values).toMatchObject({
  "--my-property": "1", // string instead of a number
});
```


- https://github.com/Fullchee/jest-dom-to-have-style/blob/main/src/App.test.js#L22-L24
